openmdao/solvers/linear/direct.py

```diff
@@ -59,7 +65,10 @@ def _linearize(self):
             eye = np.eye(nmtx)
             mtx = np.empty((nmtx, nmtx))
             for i in range(nmtx):
-                self._mat_vec(eye[:, i], mtx[:, i])
+                if self._mode == 'fwd':
```

Comment:
I don't understand why this is necessary. looping of columns or rows of identity gives the same results regardless. mtx is just an empty matrix. There should not be any difference.